src/coreclr/nativeaot/Runtime/startup.cpp

```diff
@@ -540,6 +540,10 @@ void RuntimeThreadShutdown(void* thread)
 #endif
 
     ThreadStore::DetachCurrentThread();
+    
+#ifdef FEATURE_PERFTRACING
+    EventPipe_ThreadShutdown();
+#endif
```

Comment:
This seems like a great feature that will allow EventPipe to do away with the current convoluted thread local storage management, for all the threads that gets created to manage writing events. I assume all EventPipe threads will come through here?

Reply:
Yeah, they should go through here on exit - NativeAOT already had all the infrastructure in place hooking it up. It may be a thread that never wrote any events, in which case it'll just not have any cleanup to do for event pipe.